src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -80,11 +79,10 @@ public void resolveSourceFiles(Run<?, ?> run, FilePath workspace, TaskListener l
 
         listener.getLogger().printf("%d source files need to be copied%n", paints.size());
 
-        ExecutorService service = Executors.newFixedThreadPool(5);
```

Comment:
What if instead of relying on the timer threads, we just put a timeout or add the shutdown in the finally block?